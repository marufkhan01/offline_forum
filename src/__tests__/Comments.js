import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Comments from '../components/Comments';

it('calls componentDidMount', () => {
    const spy = jest.spyOn(Comments.prototype,'componentDidMount');
    const wrapper = mount(<Comments />);
    expect(spy).toHaveBeenCalled();
});

it('call function setCommentsFromLocalStorage', () => {
    const wrapper = shallow(<Comments />);
    expect(wrapper.instance().setCommentsFromLocalStorage());
});

it('call function removeComment', () => {
    const wrapper = shallow(<Comments />);
    expect(wrapper.instance().removeComment());
});

it('veryfy values of tag', () => {
    const wrapper = shallow(<Comments />);
    expect(wrapper.find('h2').length).toBe(1);
    expect(wrapper.find('h2').at(0).text()).toBe('Comments');

});


it('Comments component should render as expected',() =>{
    const component = shallow(<Comments />);
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
});