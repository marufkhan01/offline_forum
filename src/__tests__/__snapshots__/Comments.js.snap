// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Comments component should render as expected 1`] = `
<div
  className="py-2"
>
  <h2
    className="text-indigo-darker border-b mb-4"
  >
    Comments
  </h2>
  <CreateNewComment
    updateComments={[Function]}
  />
</div>
`;
